pkg/volume/testing/testing.go

```diff
@@ -1084,14 +1084,14 @@ func (fv *FakeVolume) GetUnmountDeviceCallCount() int {
 func (fv *FakeVolume) Detach(volumeName string, nodeName types.NodeName) error {
 	fv.Lock()
 	defer fv.Unlock()
-	fv.DetachCallCount++
 
 	node := string(nodeName)
 	volumeNodes, exist := fv.VolumesAttached[volumeName]
 	if !exist || !volumeNodes.Has(node) {
 		return fmt.Errorf("trying to detach volume %q that is not attached to the node %q", volumeName, node)
 	}
 
+	fv.DetachCallCount++
```

Comment:
without understanding the mechanics of the tests, this is not counting all calls to detach, the volume  fails to be attached or doesn't exist we don't consider that call

Reply:
Yep that sounds right. The test is somewhat racy and detach can get called more than once for same volume, while we wait for detach in tests. Not counting calls of already detached volumes fixes the flake.